From fa0d63ff8c488be15976e5353580b565e85586a1 Mon Sep 17 00:00:00 2001
From: Nelson Elhage <nelhage@nelhage.com>
Date: Sat, 18 Feb 2017 16:36:40 -0800
Subject: [PATCH 1/2] Placate a gcc7 warning.

(And handle a too-long temp directory with an error instead of a likely
crash)
---
 attach.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

--- attach.c
+++ attach.c
@@ -389,8 +389,11 @@ int setup_steal_socket(struct steal_pty_state *steal) {
         return errno;
 
     steal->addr_un.sun_family = AF_UNIX;
-    snprintf(steal->addr_un.sun_path, sizeof(steal->addr_un.sun_path),
-             "%s/reptyr.sock", steal->tmpdir);
+    if (snprintf(steal->addr_un.sun_path, sizeof(steal->addr_un.sun_path),
+                 "%s/reptyr.sock", steal->tmpdir) >= sizeof(steal->addr_un.sun_path)) {
+        error("tmpdir path too long!");
+        return ENAMETOOLONG;
+    }
 
     if ((steal->sockfd = socket(AF_UNIX, SOCK_DGRAM, 0)) < 0)
         return errno;

From b45fd9238958fcf2d8f3d6fc23e6d491febea2ac Mon Sep 17 00:00:00 2001
From: Nelson Elhage <nelhage@nelhage.com>
Date: Sat, 18 Feb 2017 16:38:18 -0800
Subject: [PATCH 2/2] Include sysmacros.h

See #81
---
 platform/linux/linux.h | 1 +
 1 file changed, 1 insertion(+)

--- platform/linux/linux.h
+++ platform/linux/linux.h
@@ -40,6 +40,7 @@
 #include <sys/ptrace.h>
 #include <asm/ptrace.h>
 #include <sys/types.h>
+#include <sys/sysmacros.h>
 #include <sys/user.h>
 #include <unistd.h>
 #include <stdlib.h>
